src/Compilers/Core/MSBuildTask/Csc.cs

```diff
@@ -535,6 +535,13 @@ private bool InitializeHostCompiler(ICscHostObject cscHostObject)
                     CheckHostObjectSupport(param = nameof(AdditionalFiles), analyzerHostObject.SetAdditionalFiles(AdditionalFiles));
                 }
 
+                // For host objects which support it, set the analyzer config files and potential config files.
+                if (cscHostObject is IAnalyzerConfigFilesHostObject analyzerConfigFilesHostObject)
+                {
+                    CheckHostObjectSupport(param = nameof(AnalyzerConfigFiles), analyzerConfigFilesHostObject.SetAnalyzerConfigFiles(AnalyzerConfigFiles));
+                    CheckHostObjectSupport(param = nameof(PotentialAnalyzerConfigFiles), analyzerConfigFilesHostObject.SetPotentialAnalyzerConfigFiles(PotentialAnalyzerConfigFiles));
```

Comment:
Using the host object is the most straight-forward approach--if we're running the `Csc` or `Vbc` tasks we've already computed the sets of config files and potential config files, so we may as well pass them through.

One thing I will consider is responding to adding/removing .editorconfig files by running a design-time build that only runs the targets necessary to obtain these file sets, rather than the `Compile` target. That would be in addition to passing them through the host object, however.